src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1321,38 +1333,6 @@ fn add_library_search_dirs(cmd: &mut dyn Linker, sess: &Session) {
     cmd.include_path(&fix_windows_verbatim_for_gcc(&lib_path));
 }
 
-/// Add options requesting executables to be position-independent or not position-independent.
-fn add_position_independent_executable_args(
-    cmd: &mut dyn Linker,
-    sess: &Session,
-    flavor: LinkerFlavor,
-    crate_type: config::CrateType,
-    codegen_results: &CodegenResults,
-) {
-    if crate_type != config::CrateType::Executable {
-        return;
-    }
-
-    if sess.target.target.options.position_independent_executables {
-        let attr_link_args = &*codegen_results.crate_info.link_args;
-        let mut user_defined_link_args = sess.opts.cg.link_args.iter().chain(attr_link_args);
-        if sess.relocation_model() == RelocModel::Pic
-            && !sess.crt_static(Some(crate_type))
-            && !user_defined_link_args.any(|x| x == "-static")
```

Comment:
The only case of rustc interpreting user-provided raw linker arguments is removed here.
First, `-static` wins over `-pie` so we could still pass `-pie` if the user passed `-static` with the same effect.
Second, it was a hack, this static guessing wasn't done consistently in other places, equivalent cases like `-Wl,-static` weren't recognized, and we shouldn't be interpreting raw linking args in general.